remme/tp/pub_key.py

```diff
@@ -61,95 +70,167 @@ def get_state_processor(self):
+        if public_key_information:
+            raise InvalidTransaction('This public key is already registered.')
+
+        try:
+            certificate_public_key = load_pem_public_key(
+                new_public_key_payload.public_key.encode('utf-8'), backend=default_backend(),
+            )
+
+        except ValueError:
+            raise InvalidTransaction('Cannot deserialize the provided public key. Check if it is in PEM format.')
+
+        try:
+            ehs_bytes = binascii.unhexlify(new_public_key_payload.entity_hash_signature)
+            eh_bytes = binascii.unhexlify(new_public_key_payload.entity_hash)
+        except binascii.Error:
+            raise InvalidTransaction('Entity hash or/and signature is not a hex format.')
+
+        if not self._is_signature_valid(
+                certificate_public_key=certificate_public_key, ehs_bytes=ehs_bytes, eh_bytes=eh_bytes,
+        ):
+            raise InvalidTransaction('Invalid signature.')
+
+        certificate_valid_from, certificate_valid_to = \
+            new_public_key_payload.valid_from, new_public_key_payload.valid_to
+
+        if not self._is_certificate_validity_exceeded(valid_from=certificate_valid_from, valid_to=certificate_valid_to):
             raise InvalidTransaction('The public key validity exceeds the maximum value.')
 
-        data = PubKeyStorage()
-        data.owner = signer_pubkey
-        data.payload.CopyFrom(transaction_payload)
-        data.revoked = False
+        public_key_to_store = PubKeyStorage()
+        public_key_to_store.owner = signer_public_key
+        public_key_to_store.payload.CopyFrom(new_public_key_payload)
+        public_key_to_store.revoked = False
 
         if not account:
             account = Account()
 
-        state = {account_address: account, address: data}
-        is_economy_enabled = _get_setting_value(context,
-                                                'remme.economy_enabled',
-                                                'true').lower()
-        if is_economy_enabled == 'true':
-            storage_pub_key = _get_setting_value(context,
-                                                 SETTINGS_STORAGE_PUB_KEY)
-            if not storage_pub_key:
-                raise InvalidTransaction('The storage public key not set.')
-
-            storage_address = AccountHandler() \
-                .make_address_from_data(storage_pub_key)
-
-            if storage_address != account_address:
-                transfer_state = AccountHandler() \
-                    .create_transfer(context, account_address, storage_address,
-                                     PUB_KEY_STORE_PRICE)
+        state = {
+            sender_account_address: account,
+            public_key_to_store_address: public_key_to_store,
+        }
+
+        is_economy_enabled = _get_setting_value(context, 'remme.economy_enabled', 'true').lower()
+
+        if is_economy_enabled == ECONOMY_IS_ENABLED_VALUE:
```

Comment:
A little note about code style: we don't use an empty newline after `if` statement.

Reply:
@eugene-babichenko, got you.